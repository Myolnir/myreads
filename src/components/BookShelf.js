import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Book from './Book'

class BookShelf extends Component {

  static propTypes = {
    books: PropTypes.array.isRequired,
    moveToShelf: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired
  }

  render() {
    const { books, moveToShelf, title } = this.props
    return (
      <div className="bookshelf">
        <h2 className="bookshelf-title">{title}</h2>
        <div className="bookshelf-books">
          <ol className="books-grid">
            {books.map((book) => (
              <Book
                key={book.id}
                book={book}
                moveToShelf={moveToShelf}
              />
            ))}
          </ol>
        </div>
      </div>
    )
  }
}

export default BookShelf
