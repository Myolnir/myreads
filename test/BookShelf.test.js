import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { books }  from './book.fixture';
import BookShelf from '../src/components/BookShelf';

describe("<BookShelf />", () => {
  it("Should render the component", ()  => {
    const moveToShelf = function(){}
    const title = 'Test title'
    const bookWrapper = mount(
      <BookShelf
        books={books}
        moveToShelf={moveToShelf}
        title={title}
      />
    )
    expect(bookWrapper.html()).toMatchSnapshot();
  });

});
