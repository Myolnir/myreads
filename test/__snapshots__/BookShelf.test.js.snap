exports[`<BookShelf /> Should render the component 1`] = `"<div class=\"bookshelf\"><h2 class=\"bookshelf-title\">Test title</h2><div class=\"bookshelf-books\"><ol class=\"books-grid\"><li><div class=\"book\"><div class=\"book-top\"><div class=\"book-cover\" style=\"width: 128px; height: 193px; background-image: url(http://localhost/test.jpg);\"></div><div class=\"book-shelf-changer\"><select><option value=\"none\" disabled=\"\">Move to...</option><option value=\"currentlyReading\">Currently Reading</option><option value=\"wantToRead\">Want to Read</option><option value=\"read\">Read</option><option value=\"none\">None</option></select></div></div><div class=\"book-title\">test title</div><div class=\"book-authors\">author title</div></div></li><li><div class=\"book\"><div class=\"book-top\"><div class=\"book-cover\" style=\"width: 128px; height: 193px; background-image: url(http://localhost/test2.jpg);\"></div><div class=\"book-shelf-changer\"><select><option value=\"none\" disabled=\"\">Move to...</option><option value=\"currentlyReading\">Currently Reading</option><option value=\"wantToRead\">Want to Read</option><option value=\"read\">Read</option><option value=\"none\">None</option></select></div></div><div class=\"book-title\">test2 title</div><div class=\"book-authors\">author2 title</div></div></li></ol></div></div>"`;
